Realm/RLMQueryUtil.mm

```diff
@@ -71,6 +71,14 @@ static void RLMPrecondition(bool condition, NSString *name, NSString *format, ..
     return prop;
 }
 
+NSPredicate *RLMValidatedPredicate(id predicateFormat, va_list args) {
+    if (predicateFormat && ![predicateFormat isKindOfClass:[NSString class]]) {
```

Comment:
That's a *very* good point. Didn't we *used* to have overloads that could be either `NSString *` or `NSPredicate`? I think this is just a leftover from that. I'll revert that back.